content/en/platform/corda/5.0/application-networks/creating/members/virtual-node.md

```diff
@@ -9,17 +9,30 @@ menu:
 section_menu: corda5
 ---
 
-To create a virtual node for a member, run the following commands, changing the X.500 name:
+You can create a virtual node using the POST method of the [/api/v1/virtualnode endpoint](../../../reference/rest-api/C5_OpenAPI.html#tag/Virtual-Node-API/operation/post_virtualnode). The following sections describe how to use this method:
+* [Create a Virtual Node on Linux or macOS](#create-a-virtual-node-on-linux-or-macos)
+* [Create a Virtual Node on Windows](#create-a-virtual-node-on-linux-or-windows)
+
+## Create a Virtual Node on Linux or macOS
+
+To create a virtual node for a member on Linux or macOS, run the following commands in Bash to send the request using Curl, changing the X.500 name and using the checksum retrieved when you [uploaded the member CPI]({{< relref"./cpi.md#upload-the-cpi" >}}):
 
-{{< tabs >}}
-{{% tab name="Bash"%}}
 ```shell
 export CPI_CHECKSUM=<CPI-checksum>
 export X500_NAME="C=GB, L=London, O=Alice"
 curl -u $REST_API_USER:$REST_API_PASSWORD -d '{"request": {"cpiFileChecksum": "'$CPI_CHECKSUM'", "x500Name": "'$X500_NAME'"}}' $API_URL/virtualnode
 ```
-{{% /tab %}}
-{{% tab name="PowerShell" %}}
+
+If successful, this request returns the details of the new virtual node as JSON. To save the ID of the virtual node for future use, run the following command, replacing `<holding-identity-ID>` with the ID returned in `holdingIdentity.shortHash` in the received response. For example, `58B6030FABDD`.
+
+```shell
+export MGM_HOLDING_ID=<holding-identity-ID>
+```
+
+## Create a Virtual Node on Windows
```

Comment:
Creating a virtual node has changed to an async operation. The command already in this section will start the process of creating a vnode. The response will contain an ID representing the work of creating that vnode. We need to add an extra request to check the status of the work.

Line 45 no longer works and needs to be removed:
```powershell
$HOLDING_ID = $VIRTUAL_NODE_RESPONSE.holdingIdentity.shortHash  <- remove this
```

It is replaced with a process. This request fetches the status of the work:

```powershell
$VIRTUAL_NODE_RESPONSE_STATUS = Invoke-RestMethod -SkipCertificateCheck  -Headers @{Authorization=("Basic {0}" -f $AUTH_INFO)} -Uri "$REST_API_URL/virtualnode/status/$($VIRTUAL_NODE_RESPONSE.requestId)" -Method Get
```

This request will return a json object with `status = "SUCCEEDED"` once the operation is complete. We may need to call this multiple times until we see `SUCCEEDED`. In our own test code, we loop while the status is `ACCEPTED` or `IN_PROGRESS`. We treat `SUCCEEDED`, `FAILED` or `ABORTED` as end states and stop checking when we see them.

Once we have success, we can read the Holding ID from `resourceId` of that status response. 

```powershell
$HOLDING_ID = $VIRTUAL_NODE_RESPONSE_STATUS.resourceId
```

**At the moment, the request ID and holding ID are the same but we need to be ready for a future where they are different so we should avoid using the request ID as a holding ID in these examples.**

The bash example needs a similar update.

Reply:
@josephzunigadaly I've made a stab at this. Can you review please before I make a similar change to the MGM section. Thanks